### PR TITLE
Add Merge Sort algo chapter and Python example

### PR DESCRIPTION
**Pull request by Mukundan314**

### CODE REVIEW

**Comment by Gustorn** [comment]
This is also copy-pasted from Wikipedia which might not be what we want to do

**Comment by Butt4cak3** [comment]
Good catch, @Gustorn. I didn't even think about checking for plagiarism. That's a no-go.

**Comment by leios** [comment]
Oh yeah, a lot of work needs to be done on the text here, but I really appreciate the contribution! I am not sure the best way to handle this, but we have 3 options:

1. You keep re-writing the text until we are satisfied
2. I rewrite the text and send a PR to you
3. I rewrite the text, put it on the AAA and then you resubmit your code then

**Comment by Mukundan314** [comment]
options 2 and 3 look good to me since I am not good at explaining stuff.

**Comment by leios** [comment]
Great! Not a problem. Since I know we have code waiting for this chapter and I am sure people are interested in writing some code for it, I'll bump it up the writing list and try to get it done this weekend. Wait a few days, and I'll try to get the chapter up.

Standard procedure for chapters is to leave the chapter up for a week to make sure there are no large comments or questions and that the code is in a generally understandable state.

I will choose between the 2 options depending on how the chapter shapes up. If I can use your code, then we'll use it as the fundamental merge-sort example and I'll add Julia code to that in a separate PR. If I need to write some small functions myself for the bulk text, then I might ask for you to resubmit the code at a later date. 

Basically, give me a bit to figure out the text.

**Comment by Butt4cak3** [comment]
My suggestion would be to turn the .md file into a stub with just a title and a note that "This chapter is coming soon". We can then add the Python code import at the bottom of it and merge the entire thing. When you (@leios) write the chapter, the .md file is already there and you can add the text to it.

**Comment by leios** [comment]
Yeah, that's a good solution

File: chapters/sorting_searching/merge/merge_sort.md [hunk 1]
```diff
@@ -1,21 +1,10 @@
...
-
-In computer science, merge sort (also commonly spelled mergesort) is an efficient, general-purpose, comparison-based sorting algorithm.
-Most implementations produce a stable sort, which means that the implementation preserves the input order of equal elements in the sorted output.
-Merge sort is a divide and conquer algorithm that was invented by John von Neumann in 1945.
-A detailed description and analysis of bottom-up mergesort appeared in a report by Goldstine and von Neumann as early as 1948.
-
-Conceptually, a merge sort works as follows:
-    1. Divide the unsorted list into n sublists, each containing 1 element (a list of 1 element is considered sorted).
-    2. Repeatedly merge sublists to produce new sorted sublists until there is only 1 sublist remaining. This will be the sorted list.
+# This Chapter is comming soon
```
Comment (Butt4cak3) [inline chapters/sorting_searching/merge/merge_sort.md]: "coming" is spelled with only one "m".

**Comment by Butt4cak3** [comment]
It has been almost two months now. I'm a bit out of the loop because of my break, so @leios, what's your plan regarding stub articles like this now?

Aside from that, this PR has become "dirty". 30a5ab255e91e1e53d9433d0082d14cae83ea332 does not belong here. It is completely unrelated to what this PR is about. I assume you (@Mukundan314) accidentally committed to this PR instead of a separate branch. If you need any help removing the commit, let me know.

**Comment by leios** [comment]
Yeah, right now I clearly don't have the time to re-write this chapter... But is there any harm in leaving it up?

**Comment by Butt4cak3** [comment]
> But is there any harm in leaving it up?

You mean leaving the PR up? None at all. I just like digging up old PRs from time to time to see if we can make some progress on them.

**Comment by leios** [comment]
Yeah, no problem leaving it up. It's good to bump it so it's back on my radar.

**Comment by ntindle** [comment]
[lang: python]
